packages/mui-material/src/styles/CssVarsProvider.js

```diff
@@ -0,0 +1,26 @@
+import { unstable_createCssVarsProvider as createCssVarsProvider } from '@mui/system';
+import unstable_createTheme from './unstable_createTheme';
+import createTypography from './createTypography';
+
+const defaultTheme = unstable_createTheme();
+
+const { CssVarsProvider, useColorScheme, getInitColorSchemeScript } = createCssVarsProvider({
+  theme: defaultTheme,
+  defaultColorScheme: {
+    light: 'light',
+    dark: 'dark',
+  },
+  prefix: 'md',
+  resolveTheme: (theme) => {
+    const newTheme = {
+      ...theme,
+      typography: createTypography(theme.palette, theme.typography),
+    };
+
+    return newTheme;
+  },
```

Comment:
```suggestion
```
From what I checked in `createTypography`, there is no need to recalculate here. Developer can customize the typography via `experimental_extendTheme`.

Reply:
I don't think we can, as the typography is being generated by the palette, which based on the `colorScheme` set on the CssVarsProvider is being calculated.

Reply:
Okay, let's keep it like this but I think we should be able to remove it in the future. In `createTypography.js`, I don't see any usage of the palette internally. With the CSS variables, I don't see a need to accept typography as function because developer can use the palette via CSS variables.

```js
extendTheme({
  typography: {
    h1: {
      color: 'var(--joy-palette-text-primary)',
    }
  }
})
```